pkg/daemon/daemon.go

```diff
@@ -1629,3 +1634,71 @@ func forceFileExists() bool {
 
 	return false
 }
+
+// patchLabelsOnNode takes a map of key/value string pairs and patches them as labels on the node
+func (dn *Daemon) patchLabelsOnNode(labels map[string]interface{}) error {
+	var err error
+	var patchData []byte
+	patch := struct {
+		Metadata map[string]interface{} `json:"metadata"`
+	}{
+		Metadata: map[string]interface{}{
+			"labels": labels,
```

Comment:
Thanks for your feedback! Can you please provide an example of how you see it? (I suppose it's straight forward, but at this hour I don't see it... :) )

The reason I use `labels map[string]interface{})` as the input type and later in the struct is that I need to pass it `nil` when I want to remove the label. Would I be able to do it with `Labels map[string]string` from `ObjectMeta`?

Reply:
I attempted to do this here: https://github.com/openshift/machine-config-operator/pull/2954 , but the semantics of nil in the merge patch don't carry through with the switch to ObjectMeta.

Reply:
> If not, labels is missing the json:"labels" tag and this could cause problems.

Because this is a map, the literal "labels" is used in the json.

Reply:
it's not obvious, but there's a "normal" way to do this in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/label/label.go#L338 .  I would swap to it, but without an e2e test, the verification is harder for me.

Reply:
> Because this is a map, the literal "labels" is used in the json.

That's a great point. Thanks for the correction!

> I attempted to do this here: https://github.com/openshift/machine-config-operator/pull/2954 , but the semantics of nil in the merge patch don't carry through with the switch to ObjectMeta.

Between the nil semantics and your comment about the `labels` literal, I'll consider this resolved as-is.